### PR TITLE
Adding google GCE

### PR DESCRIPTION
**Pull request by CorbanR**
- Adding google gce roles
- Adding custom gce.py
- Adding library path to ansible.cfg
- Updating Tunnelblick url, and version bump
- Updating Shadowsocks-win-dotnet url and version bump
- Updating ShadowsocksX url and version bump

### CODE REVIEW

File: playbooks/google.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+  # This variable file is included so the gce-network role knows
+  # which port to open for SSH
+  vars_files:
+    - roles/ssh/defaults/main.yml
+
+  vars_prompt:
+    - name: "gce_zone_var"
+      prompt: >
+        What zone should the server be located in?
```
Comment (jlund) [inline playbooks/google.yml]: Small thing, but can we make all the parentheses line up? We also only need four spaces after "United States" in order to be consistent with the Amazon configuration. This should make the output look a little cleaner in the terminal.

File: playbooks/google.yml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          1. United States        (Datacenter A)
```
Comment (jlund) [inline playbooks/google.yml]: Let's say "Iowa A" instead of "Datacenter A".

File: playbooks/google.yml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          2. United States        (Datacenter B)
+          3. United States        (Datacenter C)
+          4. United States        (Datacenter F)
+          5. Europe         (Datacenter B)
```
Comment (jlund) [inline playbooks/google.yml]: "Datacenter B" -> "Belgium B", etc.

File: playbooks/google.yml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          6. Europe         (Datacenter C)
+          7. Europe         (Datacenter D)
+          8. Asia         (Datacenter A)
```
Comment (jlund) [inline playbooks/google.yml]: "Asia" -> "East Asia"
"Datacenter A" -> "Taiwan A"

File: playbooks/google.yml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      default: "3"
+      private: no
+
+    - name: "gce_server_name_var"
+      prompt: "\nWhat should the server be named? Press enter for default (streisand).\n"
+      default: "streisand"
+      private: no
+
+    - name: "gce_pem_file_var"
+      prompt: "\n\nThe location of your unique gce pem file. Details on generating this can be found at \nhttp://docs.ansible.com/guide_gce.html#credentials\n"
```
Comment (jlund) [inline playbooks/google.yml]: "gce" -> "Google Compute Engine"
"pem" -> "PEM"

File: playbooks/google.yml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      default: "~/.ssh/google.pem"
+      private: no
+
+    - name: "gce_project_id_var"
+      prompt: "\nWhat is your GCE project id?\nThis can be found https://console.developers.google.com/project\n"
```
Comment (jlund) [inline playbooks/google.yml]: "found" -> "found at"

File: playbooks/google.yml [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      private: no
+
+    - name: "gce_service_account_email_var"
+      prompt: "\n\nWhat is your GCE service account email?\nThis can be found https://console.developers.google.com/project/<your_project_id>/apiui/credential\n"
```
Comment (jlund) [inline playbooks/google.yml]: "found" -> "found at"

File: playbooks/google.yml [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    l2tp_ipsec_firewall_rules: ""
+
+    # Similarly, we don't want to display a link to nonexistent
+    # L2TP/IPsec information on the Gateway index page
+    l2tp_ipsec_gateway_location: ""
+
+  roles:
+    - common
+    - openvpn
+    - dante
```
Comment (jlund) [inline playbooks/google.yml]: This role no longer exists. See #157 for more details. Hooray for faster setup!

File: playbooks/roles/genesis-google/vars/main.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (jlund) [inline playbooks/roles/genesis-google/vars/main.yml]: These are getting set in the main playbook too. So we probably don't need them defined here as well.

This is reminding me (again) that I need to refactor how to turn L2TP/IPsec off for Amazon and GCE. A parameterized include with a conditional is a much better way to disable L2TP/IPsec. That will probably be one of the first things I do after this is merged.

**Comment by jlund** [comment]
This is looking fantastic. I'm really excited to get this merged.

**Comment by CorbanR** [comment]
Alright made updates and verified that GCE works. :)

**Comment by jlund** [comment]
Works like a charm. Thank you! This is really great.

**Pull request merged**